experiments/dataSources/SalesforceREST/EinsteinAnalytics/readme.md

```diff
@@ -0,0 +1,95 @@
+- [Analytics query language (SAQL) reference](https://developer.salesforce.com/docs/atlas.en-us.bi_dev_guide_saql.meta/bi_dev_guide_saql/bi_saql_intro.htm)
+
+
+## How to use this data source
+
+### Create the data source
+
+Follow the instructions in the "SalesforceREST" folder immediately above to create a generic "Salesforce REST" datasource. 
+
+### Create a new page (V2 page API)  and copy the XML from the file:  "Einstein_Analytics_Resources"  
+
+This will show you all the datasets and lenses you have enabled in the org and give you direct links to them. 
+
+## But I want some Data! 
+
+Showing Data from Einstein Analytics requires use of the [Query Endpoint](https://developer.salesforce.com/docs/atlas.en-us.bi_dev_guide_rest.meta/bi_dev_guide_rest/bi_resources_query.htm)
+
+Here is how: 
+
+#### Find a resource in Einstein Analytics studio - prepare it for use in a query
+
+-   Open a lens or dashboard element.  Or create a new one! 
+-   Open the "Query mode" view of a lens,  or use the "more details" button on the details panel of a dashboard.  Look for the query code.  It looks like this: 
+
+```
+q = load "DTC_Opportunity_SAMPLE";
+q = group q by ('Industry', 'Product_Family');
+q = foreach q generate 'Industry' as 'Industry', 'Product_Family' as 'Product_Family', sum('Amount') as 'sum_Amount';
+q = order q by 'sum_Amount' desc;
+q = limit q 2000; 
+```
+    
+- Copy this code to a text editor and make the following changes: 
+    -  Add a JSON wrapper to the query.  
+        - Append Append ``{"query":"``  to the beginning
+        - Finish block with ``"}``
+
+    - Replace the name of the dataset (that comes in the first line after “load” with a combination of its id and current version id.  You can find the Id’s in the EA_Resources page that we created above. 
+
+    - Add a leading `\` in front of any double quote in the query statement itself.  (Don’t replace the double quotes in the JSON wrapper you added in step 1. 
+
+    - Take out all line breaks.  The JSON block will be one long line.   It will look somthing like this: 
+
+``{"query":"q = load \"0Fb6g0000008aw2CAA/0Fc6g0000009ZKoCAM\";q = group q by (‘Close_Date_Year’, ‘Account_Type’);q = foreach q generate ‘Close_Date_Year’ as ‘Close_Date_Year’, ‘Account_Type’ as ‘Account_Type’, sum('Amount') as 'sum_Amount';q = order q by 'sum_Amount' desc;q = limit q 2000;"}``
+
+
+#### Create a new REST model on your Skuid page
+
+Properties: 
+- Data Source URL:  “/services/data/v42.0/wave/query”
+- HTTP Verb:  “POST” 
+- Request body contents “As Templated Request Body”
+- Template:  Copy the JSON block you created above. 
+
+- You will need to swap over to the XML and add a “Path to Contents” entry in the highest level node of the model.   Add this to the end of the node  ``<model id=” …    pathtocontent="results.records" ``
```

Comment:
Since we're exiting the properties pane for this, having it as the same list as the items above is confusing. We could do a few things:

- Make this "Swap over to the XML" section not a list item, but then you'd have to do the same for manually creating fields. Maybe you could make headers for both of them?
- Make the `Properties` line above a list item, with each key/value as a sub-list to that item. Probably lowest LOE way to solve this.
```suggestion
- You will need to swap over to the XML and add a “Path to Contents” entry in the highest level node of the model.   Add this to the end of the node  ``<model id=” …    pathtocontent="results.records" ``
```